src/algorithms/diamondsquare.jl

```diff
@@ -0,0 +1,210 @@
+"""
+    DiamondSquare
+
+This type generates a neutral landscape using the diamond-sqaures
+algorithm, which produces fractals with variable spatialautocorrelation.
+
+https://en.wikipedia.org/wiki/Diamond-square_algorithm
+
+The algorithm is named diamond-square because it is an iterative procedure of
+"diamond" and "square" steps.
+
+
+
+The degree of spatial autocorrelation is controlled by a parameter `H`,
+which varies from 0.0 (low autocorrelation) to 1.0 (high autocorrelation) --- note this is non-inclusive and H = 0 and H = 1 will not behavive as expected.
+The result of the diamond-square algorithm is a fractal with dimension D = 2 + H.
+
+A similar algorithm, midpoint displacement *TODO link to mpd in docs*, almost
+identically, except that in DiamondSquare, the ed
+"""
+struct DiamondSquare <: NeutralLandscapeMaker
+    H::Float64
+    DiamondSquare(H::T) where {T <: Real} = new(H-floor(H)) # enforce H ∈ (0,1)
```

Comment:
Don't we want to error on H outside of 0,1?

Reply:
Absolutely -- a simple `@assert 0.0 < H <= 1.0` would work here